lock/src/main/java/org/infinispan/lock/impl/lock/ClusteredLockImpl.java

```diff
@@ -170,19 +202,25 @@ protected void handle(Boolean result) {
                // In this case, even if the lock was marked as acquired in the cache, it has to be released because the call expired.
                // We have to unlock the lock if the requestor and the requestId match.
                // Meanwhile another request for this owner might have locked it successfully and we don't want to unlock in that case
-               unlock(requestId, requestor);
+               unlock(requestId, Collections.singleton(requestor));
             }
-         } else if(!isScheduled){
+         } else if (!isScheduled) {
+            log.tracef("Schedule for expiration %s", this);
             // If the lock was not acquired, then schedule a complete false for the given timeout
             isScheduled = true;
             clusteredLockManager.schedule(() -> request.complete(false), time, unit);
```

Comment:
We should log a JIRA to clean up old scheduled tasks when the request is complete as well.

Reply:
Is already done ! https://issues.jboss.org/browse/ISPN-8757